File: exercises/02 photosearcher/cs/PhotoSearcher/PhotoDeleter.cs
using System;
using System.Collections.Generic;

namespace PhotoSearcher
{
    public class PhotoDeleter
    {
        private IPhotoRepository _photoRepository;

        public PhotoDeleter(IPhotoRepository photoRepository)
        {
            _photoRepository = photoRepository;
        }

        public void DeletePhotos(FuzzyDate fuzzyDate)
        {
            // Use test-driven development to implement this method
            throw new NotImplementedException();
        }

        public void DeletePhotos(FuzzyDate fuzzyDate, Color color, ContentType? contentType)
        {
            // Use test-driven development to implement this method
            throw new NotImplementedException();
        }

    }
}

File: solutions/02 photosearcher/java/src/eu/sioux/tdd/photosearcher/PhotoDeleter.java
package eu.sioux.tdd.photosearcher;

import sun.reflect.generics.reflectiveObjects.NotImplementedException;

import java.util.List;
import java.util.Optional;

public class PhotoDeleter {
    private final PhotoRepository photoRepository;

    public PhotoDeleter(PhotoRepository photoRepository) {
        this.photoRepository = photoRepository;
    }

    public void deletePhotos(FuzzyDate date) {
        // Use test-driven development to implement this method
        throw new NotImplementedException();
    }

    public void deletePhotos(Optional<FuzzyDate> fuzzyDate, Optional<Color> color, Optional<ContentType> contentType) {
        // Use test-driven development to implement this method
        throw new NotImplementedException();
    }

}